arcgispro/3D Data Co-Registration.pyt

```diff
@@ -322,17 +321,48 @@ class Register_MultiType(object):
                             "Consider reprojecting input DEM"
                             f" X = {band_description.meanCellWidth}, Y = {band_description.meanCellHeight}"
                         )
+        
+        #check that correct input 3D data types are being used ("las", "laz", "bpf", "ply", "obj", "tif", "tiff")
+        acceptable_data_list = [".las", ".laz", ".bpf", ".ply", ".obj", ".tif", ".tiff"]
+        if parameters[0].value:
+            fnd_full_path = os.fsdecode(f"{self.getLayerPath(parameters[0].valueAsText)}").replace(os.sep, "/")
+            fnd_file_extension = os.path.splitext(fnd_full_path)[-1]
+            if fnd_file_extension not in acceptable_data_list:
+                parameters[0].setErrorMessage(
+                    "File not able to be coregistered."
+                    f" Acceptable file types are {acceptable_data_list}"
+                )
+        if parameters[1].value:
+            aoi_full_path = os.fsdecode(f"{self.getLayerPath(parameters[1].valueAsText)}").replace(os.sep, "/")
+            aoi_file_extension = os.path.splitext(aoi_full_path)[-1]
+            if aoi_file_extension not in acceptable_data_list:
+                parameters[1].setErrorMessage(
+                    "File not able to be coregistered."
+                    f" Acceptable file types are {acceptable_data_list}"
+                )
 
         return
 
+    def getLayerPath(self, layer):
+        if os.path.isabs(layer):
+            layer_source = layer
+        else:
+            desc = arcpy.Describe(layer)
+            path = desc.path
+            layer_source = str(path) + "\\" + layer
```

Comment:
if `path` represents a path (C:\\Users....) we should stick w/ `os.path.join` calls here

```python
layer_source = os.path.join(desc.path, layer)
```

Reply:
incorporated!